source/frontend/src/components/common/ApiVersionInfo.tsx

```diff
@@ -12,11 +13,11 @@ export const ApiVersionInfo = () => {
   const { getVersion } = useApiHealth();
   const [version, setVersion] = React.useState<IApiVersion>();
 
-  React.useEffect(() => {
+  useDeepCompareEffect(() => {
     let isActive = true;
     const get = async () => {
       const response = await getVersion();
-      if (isActive) {
+      if (isActive && version?.informationalVersion !== response.data?.informationalVersion) {
```

Comment:
Can you elaborate on why is this change necessary? I am guessing that fixing the tests uncovered some unwanted behavior on the app.

Reply:
Yes, in some cases the test uncovered situations where the getVersion request would get fired more then once. In that case, we don't need to overwrite the version information unless it has changed.